go/libkb/saltpack.go

```diff
@@ -38,6 +38,29 @@ func (b naclBoxPublicKey) HideIdentity() bool {
 	return false
 }
 
+type hiddenNaclBoxPublicKey NaclDHKeyPublic
+
+func (b hiddenNaclBoxPublicKey) ToKID() []byte {
+	return b[:]
```

Comment:
I can't guarantee that here (maybe some other code changes in the future would break it). I would really prefer to "subclass" `naclBoxPublicKey`, but I couldn't find a clean way to do that in Go.